packages/iris-grid/src/IrisGridPartitionedTableModel.ts

```diff
@@ -0,0 +1,96 @@
+/* eslint class-methods-use-this: "off" */
+import type {
+  Column,
+  dh as DhType,
+  PartitionedTable,
+  Table,
+} from '@deephaven/jsapi-types';
+import { Formatter } from '@deephaven/jsapi-utils';
+import { ColumnName } from './CommonTypes';
+import IrisGridModel from './IrisGridModel';
+import IrisGridTableModel from './IrisGridTableModel';
+import EmptyIrisGridModel from './EmptyIrisGridModel';
+
+export function isIrisGridPartitionedTableModel(
+  model: IrisGridModel
+): model is IrisGridPartitionedTableModel {
+  return (
+    (model as IrisGridPartitionedTableModel).partitionedTable !== undefined
+  );
+}
+
+class IrisGridPartitionedTableModel extends EmptyIrisGridModel {
+  readonly partitionedTable: PartitionedTable;
+
+  private partitionKeys: unknown[];
+
+  /**
+   * @param dh JSAPI instance
+   * @param table Partitioned table to be used in the model
+   * @param formatter The formatter to use when getting formats
+   */
+  constructor(
+    dh: DhType,
+    partitionedTable: PartitionedTable,
+    formatter = new Formatter(dh)
+  ) {
+    super(dh, formatter);
+    this.partitionedTable = partitionedTable;
+    // Get the most recently added key
+    let lastKey;
+    this.partitionedTable.getKeys().forEach(key => {
+      lastKey = key;
+    });
+    this.partitionKeys = Array.isArray(lastKey) ? lastKey : [lastKey];
+  }
+
+  get isPartitionRequired(): boolean {
+    return true;
+  }
+
+  get isReversible(): boolean {
+    return false;
+  }
+
+  displayString(
+    value: unknown,
+    columnType: string,
+    columnName?: ColumnName
+  ): string {
+    return '';
+  }
+
+  close(): void {
+    this.partitionedTable.close();
+  }
+
+  get partitionColumns(): readonly Column[] {
+    return this.partitionedTable.keyColumns;
+  }
+
+  get partition(): unknown[] {
+    return this.partitionKeys;
+  }
+
+  set partition(partition: unknown[]) {
+    this.partitionKeys = partition;
+  }
+
+  get partitionKeysTable(): Promise<Table> {
+    const sorts = this.partitionColumns.map(column => column.sort().desc());
+    return this.partitionedTable.getKeyTable().then(table => {
+      table.applySort(sorts);
+      return table;
+    });
+  }
```

Comment:
We shouldn't be doing an expensive operation like this/creating a table for a property accessor. Property accessors should be either just quick reads or memoized results, so that accessing the property multiple times is consistent. Right now, this will create a table every time you read this property, which is not what you're expecting; e.g if you did something like:

```
if (model.partitionKeysTable) {
  setNextModel(createModel(model.partitionKeysTable))
}
```
You'd have a table created in the `if` expression that would never be closed/cleaned up properly. And actually, in `IrisGrid.tsx` you do exactly that, leaking a new table on each re-render:
```
           {model.partitionKeysTable && (  // Creating one key table here
                <IrisGridPartitionSelector
                  dh={model.dh}
                  tablePromise={model.partitionKeysTable} // Creating another key table here that is getting passed into IrisGridPartitionSelector
```
Instead, this should be a method so it's explicitly clear you're creating a new object. I would mimick `valuesTable` - basically just remove the `get `